db/migrate/20170503172351_add_web_domain_to_accounts.rb

```diff
@@ -0,0 +1,16 @@
+class AddWebDomainToAccounts < ActiveRecord::Migration[5.0]
+  def up
+    add_column :accounts, :web_domain, :string, null: true, default: nil
+
+    ActiveRecord::Base.transaction do
```

Comment:
You are absolutely right that this migration script is impossible to run. It will take hours if not days, and migrations place a lock on the entire schema meaning it would be hours of downtime. I am almost veering towards your initial approach of just iterating over everything.